src/GuiBase/Utils/KeyMappingManager.cpp

```diff
@@ -7,73 +7,113 @@ namespace Gui {
+        LOG( logINFO ) << "try to get action from an invalid context";
+        return KeyMappingAction();
     }
-    m_mapping[action] = keyCode;
-}
+    // skip key as modifiers,
+    if ( ( key == Qt::Key_Shift ) || ( key == Qt::Key_Control ) || ( key == Qt::Key_Alt ) ||
+         ( key == Qt::Key_Meta ) )
+    { key = -1; } KeyMappingManager::MouseBinding binding{buttons, modifiers, key, wheel};
 
-int KeyMappingManager::getKeyFromAction( KeyMappingAction action ) {
-    return m_mapping[action];
-}
+    auto action = m_mappingAction[context].find( binding );
+    if ( action != m_mappingAction[context].end() ) { return action->second; }
 
-bool KeyMappingManager::actionTriggered( QMouseEvent* event, KeyMappingAction action ) {
-    return ( int( event->buttons() ) | event->modifiers() ) == getKeyFromAction( action );
+    return KeyMappingManager::KeyMappingAction();
 }
 
-bool KeyMappingManager::actionTriggered( QKeyEvent* event, KeyMappingAction action ) {
-    return ( event->key() | event->modifiers() ) == getKeyFromAction( action );
+void KeyMappingManager::bindKeyToAction( Ra::Core::Utils::Index contextIndex,
+                                         const MouseBinding& binding,
+                                         Ra::Core::Utils::Index actionIndex ) {
+
+    CORE_ASSERT( contextIndex < m_contextNameToIndex.size(), "contextIndex is out of range" );
+
+    auto f = m_mappingAction[contextIndex].find( binding );
+    if ( f != m_mappingAction[contextIndex].end() )
+    {
+
+        auto findResult = std::find_if(
+            std::begin( m_actionNameToIndex[contextIndex] ),
+            std::end( m_actionNameToIndex[contextIndex] ),
+            [&]( const ActionNameMap::value_type& pair ) { return pair.second == actionIndex; } );
+        if ( findResult == std::end( m_actionNameToIndex[contextIndex] ) )
+        {
+            LOG( logERROR ) << "Corrupted call to bindKeyToAction index " << actionIndex
+                            << " must have been "
+                               "inserted before !\n";
+        }
+        auto test = findResult->first;
+
+        auto findResult2 = std::find_if(
+            std::begin( m_actionNameToIndex[contextIndex] ),
+            std::end( m_actionNameToIndex[contextIndex] ),
+            [&]( const ActionNameMap::value_type& pair ) { return f->second == actionIndex; } );
+
+        LOG( logWARNING ) << "Binding action " << findResult->first << " to "
+                          << "buttons [" << enumNamesFromMouseButtons( binding.m_buttons ) << "] "
+                          << "modifiers [" << enumNamesFromKeyboardModifiers( binding.m_modifiers )
+                          << "] "
+                          << "keycode [" << binding.m_key << "]"
+                          << "wheel [" << binding.m_wheel << "]"
+                          << ", which is already used for action " << findResult2->first << ".";
```

Comment:
Please also check the validity of `findResult2` before using it

Reply:
Done (btw this code was erroneous ... some bad copy/paste).